### PR TITLE
chore(pie-webc-core): DSW-1488 add lit as a dependency

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Review by ashleynolan — APPROVED** [review]
🎉

File: packages/tools/pie-icons-webc/vite.config.js [hunk 1]
```diff
@@ -30,7 +30,6 @@ export default defineConfig({
             fileName: () => 'dist/[name].js',
         },
         rollupOptions: {
-            external: [/^lit/],
```
Comment (xander-marjoram) [inline packages/tools/pie-icons-webc/vite.config.js:33]: This wasn't doing anything wrong, it just wasn't necessary. The icons-webc package just builds components so doesn't use lit directly. I checked the icon bundle sizes and they stayed the same so should be okay to remove. I also tested in the vanilla app by removing everything except an icon and it works just fine with no other dependencies.

File: packages/components/pie-webc-core/package.json [hunk 2]
```diff
@@ -3,16 +3,20 @@
...
   "scripts": {
+    "build": "run -T vite build",
     "lint:scripts": "run -T eslint .",
     "lint:scripts:fix": "run -T eslint . --fix",
     "test": "run -T vitest run",
     "test:ci": "yarn test",
     "test:watch": "run -T vitest"
   },
+  "dependencies": {
+    "lit": "3.1.0"
```
Comment (raoufswe) [inline packages/components/pie-webc-core/package.json:18]: shall we add `@lit/react` as well? 
I had a quick chat with @ashleynolan before and we believe it shouldn't affect the bundle size if users don't import the react wrapper and will eventually be three shaken,

**Review by raoufswe — APPROVED** [review]
🏅

**Review by ashleynolan — APPROVED** [review]

File: packages/components/pie-webc-core/package.json (hunk 2 above)
Comment (xander-marjoram) [reply packages/components/pie-webc-core/package.json:18]: Looks like this will be non-trivial so I won't include it as part of this PR

**Pull request merged**